notebooks/Manage Map Areas with Group and Index.ipynb

```diff
@@ -0,0 +1,456 @@
+    "\n",
+    "        # use group .notify method to send email - message supports html\n",
+    "        notify_result = notify_group.notify(users=notification_list, subject=subject, message=message,method=\"email\")\n",
+    "        \n",
+    "        return notify_result[\"success\"]\n",
+    "        \n",
+    "    except:\n",
+    "        return False"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def create_email_report(map_areas_hfl_item_id, processing_start_time):\n",
+    "    \n",
+    "    try:\n",
+    "        # create html table that includes the query results from Manage Map Area Processing table\n",
+    "        email_subject = \"\"\n",
+    "        email_body = \"\"\"<table border='1'>\n",
+    "                            <tr>\n",
+    "                                <th>web map</th>\n",
+    "                                <th>map area name</th>\n",
+    "                                <th>processing status</th>\n",
+    "                                <th>processing seconds</th>\n",
+    "                                <th>total storage mb</th>\n",
+    "                            </tr>\"\"\"\n",
+    "        out_fields = \"web_map_name, map_area_name,process_status,process_date,process_seconds,total_storage_mb\"\n",
+    "\n",
+    "        # get logging table\n",
+    "        map_areas_hfl = gis.content.get(map_areas_hfl_item_id)\n",
+    "        logging_table = map_areas_hfl.tables[0]\n",
+    "    \n",
+    "        # email subject - success count \n",
+    "        where_clause = f\"process_status = 'success' and process_date > '{processing_start_time}'\"\n",
+    "        results = logging_table.query( where = where_clause, return_count_only=True )\n",
+    "        if results > 0:\n",
+    "            email_subject = f\"{results} map areas created\"\n",
+    "        \n",
+    "        # email subject - failed count\n",
+    "        where_clause = f\"process_status = 'failed' and process_date > '{processing_start_time}'\"\n",
+    "        results = logging_table.query(where = where_clause,return_count_only=True)    \n",
+    "        if results > 0:\n",
+    "            # check to see if email subject has text - this is just for formatting\n",
+    "            if len(email_subject) > 0: \n",
+    "                email_subject += f\" / {results} map areas failed\"\n",
+    "            else:\n",
+    "                email_subject += f\"{results} map areas failed\"\n",
+    "            \n",
+    "        # email body - assemble html table rows     \n",
+    "        where_clause = f\"process_date > '{processing_start_time}'\"  \n",
+    "        results = logging_table.query(where = where_clause,out_fields = out_fields,sort_field=\"process_date\", sort_order=\"asc\")\n",
+    "        for row in results.features: \n",
+    "            if row.attributes[\"process_status\"] == \"failed\":\n",
```

Comment:
Since only the style is different, you might be able to simplify this if block to something like
```
            row_style = "style='color:#FF0000'" if row.attributes["process_status"] == "failed" else ""
            email_body += f"""<tr {row_style}>
                                <td>{row.attributes["web_map_name"]}</td>
                                <td>{row.attributes["map_area_name"]}</td>
                                <td>{row.attributes["process_status"]}</td>
                                <td>{row.attributes["process_seconds"]}</td>
                                <td>{row.attributes["total_storage_mb"]}</td>
                            </tr>"""
```

Reply:
Great suggestion - updated